Add dependabot and renovate documentation page (#6236)
astral-sh/uv: docs/guides/integration/dependency-bots.md

# Dependency bots

## Renovate

uv is supported by [Renovate](https://github.com/renovatebot/renovate).

> With `lockFileMaintenance` enabled, Renovate will automatically detect `uv.lock` and suggest updates
> to `uv.lock` files.
> 
> ```json5 title="renovate.json5"
> {
>   $schema: "https://docs.renovatebot.com/renovate-schema.json",
>   lockFileMaintenance: {
>     enabled: true,
>   },
> }
> ```

Reviewer: `lockFileMaintenance` is not required for uv support, it mostly allows refreshing transitive dependencies (or also direct dependencies if using range of versions for dependencies in `pyproject.toml`) in the lock file on a regular basis, so this should probably be a dedicated paragraph.

I'd probably go with something like:

```suggestion
Renovate will detect the presence of `uv.lock` file to determine that uv is used for managing
dependencies, and will suggest updates to
both [project dependencies](../../concepts/dependencies.md#project-dependencies), [optional dependencies](../../concepts/dependencies.md#optional-dependencies)
and [development dependencies](../../concepts/dependencies.md#development-dependencies), updating
`uv.lock` file along the way.

It is also possible to refresh `uv.lock` file on a regular basis (for instance to update transitive
dependencies) by
enabling [`lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)
option:

```json5 title="renovate.json5"
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  lockFileMaintenance: {
    enabled: true,
  },
}
```
```

Author: Thanks!

!!! note

>     `uv pip compile` outputs such as `requirements.txt` are not yet supported by Renovate.

Reviewer: Created a dedicated GitHub discussion for that (https://github.com/renovatebot/renovate/discussions/30899) that will hopefully be converted to an issue, to better track the implementation. Will report back if it ends up being converted to an issue.

Reviewer: An issue now exists: https://github.com/renovatebot/renovate/issues/30909.

Author: Awesome, I'll link to that.

# Dependency bots

## Renovate

uv is supported by [Renovate](https://github.com/renovatebot/renovate).

Renovate uses the presence of a `uv.lock` file to determine that uv is used for managing
dependencies, and will suggest upgrades to
[project dependencies](../../concepts/dependencies.md#project-dependencies),
[optional dependencies](../../concepts/dependencies.md#optional-dependencies) and
[development dependencies](../../concepts/dependencies.md#development-dependencies). Renovate will
update both the `pyproject.toml` and `uv.lock` files.

> The lockfile on also be refreshed on a regular basis (for instance to update transitive

Reviewer: ```suggestion
The lockfile can also be refreshed on a regular basis (for instance to update transitive
```

dependencies) by enabling the
[`lockFileMaintenance`](https://docs.renovatebot.com/configuration-options/#lockfilemaintenance)
option:

```json5 title="renovate.json5"
{
  $schema: "https://docs.renovatebot.com/renovate-schema.json",
  lockFileMaintenance: {
    enabled: true,
  },
}
```

!!! note

    `uv pip compile` outputs such as `requirements.txt` are not yet supported by Renovate.
    Progress can be tracked at [renovatebot/renovate#30909](https://github.com/renovatebot/renovate/issues/30909).

## Dependabot

> Support for uv  is not yet available. Progress can be tracked at

Reviewer: ```suggestion
Support for uv is not yet available. Progress can be tracked at
```
